Extending a tsconfig.json with a dictionary fails

Hi,

I have the following in a root `BUILD.bazel` file:

```bazel
load("@aspect_rules_ts//ts:defs.bzl", "ts_project")
load("@npm//:defs.bzl", "npm_link_all_packages")

ts_project(
    name = "index",
    srcs = [
        "index.ts",
    ],
    declaration = True,
    extends = ":tsconfig.json",
    tsconfig = {},
)

npm_link_all_packages(name = "node_modules")
```

On running `bazel build //:index` I get an error:

```
tsconfig_index.json(1,136): error TS6053: File 'tsconfig.json' not found.
```

Looking in the `bin` everything does seem to be in order, so I'm wondering if the working directory is incorrect maybe?

--- a/examples/simple/BUILD.bazel
+++ b/examples/simple/BUILD.bazel
@@ -39,6 +39,7 @@ ts_project(
         "foo.ts",
         "generated.ts",
     ],
+    extends = "tsconfig.json",
     # A tsconfig.json file will be generated with this content
     tsconfig = {
         "compilerOptions": {

--- a/ts/private/ts_config.bzl
+++ b/ts/private/ts_config.bzl
@@ -71,10 +71,14 @@ def _write_tsconfig_rule(ctx):
     # TODO: is it useful to expand Make variables in the content?
     content = "\n".join(ctx.attr.content)
     if ctx.attr.extends:
-        content = content.replace(
-            "__extends__",
-            relative_file(ctx.file.extends.short_path, ctx.outputs.out.short_path),
-        )
+        # Unlike other paths in the tsconfig file, the "extends" property
+        # is documented: "The path may use Node.js style resolution."
+        # https://www.typescriptlang.org/tsconfig#extends
+        # That means that we must start with explicit "./" segment.
+        extends_path = relative_file(ctx.file.extends.short_path, ctx.outputs.out.short_path)
+        if not extends_path.startswith("../"):
+            extends_path = "./" + extends_path
+        content = content.replace("__extends__", extends_path)
 
     filtered_files = _filter_input_files(ctx.files.files, ctx.attr.allow_js, ctx.attr.resolve_json_module)
     if filtered_files:

